python/fledge/plugins/north/azure_iot/azure_iot.py

```diff
@@ -89,74 +92,62 @@ def plugin_reconfigure():
-                read["asset"] = p['asset_code']
-                read["readings"] = p['reading']
-                read["timestamp"] = p['user_ts']
-                
+                read = {"asset": p['asset_code'], "readings": p['reading'], "timestamp": p['user_ts']}
                 size_payload_block += sys.getsizeof(json.dumps(read, separators=(',', ':')).encode('utf-8'))
-                if size_payload_block > MESSAGE_SIZE_LIMIT * 0.9: 
+                if size_payload_block > message_size_limit * 0.9:
                     # less than 90% of the maximum value is a quick solution to catch the not yet calculated
                     # overhead of the message class. Will be improved in a future version
                     _LOGGER.warning("The size of the message is larger than 256 kB! "
                                     "The remaining payloads will be sent on the next function call")
                     break
-                
                 last_object_id = p["id"]
                 payload_block.append(read)
-                
             num_sent = await self._send_payloads(payload_block)
             is_data_sent = True
         except Exception as ex:
-            _LOGGER.exception("Data could not be sent, %s", str(ex))
-
+            _LOGGER.exception(ex, 'Failed on sending payload.')
         return is_data_sent, last_object_id, num_sent
 
     async def _send_payloads(self, payload_block):
-        """ send a list of block payloads"""
-        
+        """send a list of block payloads"""
         num_count = 0
+        exception_raised = False
         try:
             device_client = IoTHubDeviceClient.create_from_connection_string(
-                config["primaryConnectionString"]["value"], websockets=config["websockets"]["value"])
-
+                self.primary_connection_string, websockets=self.mqtt_over_websocket)
             # Connect the device client.
             await device_client.connect()
-            
-            await self._send(device_client, payload_block)
+            # separators used only to get minimum payload size to remove whitespaces around ',' ':'
+            payload = json.dumps(payload_block, separators=(',', ':')).encode('utf-8')
+            message = Message(data=payload, content_encoding="utf-8", content_type="application/json")
+            _LOGGER.debug("Sending message: {} and of size: {}".format(message, str(message.get_size())))
+            await device_client.send_message(message)
         except Exception as ex:
-            _LOGGER.exception(f'Exception sending payloads: {ex}')
+            exception_raised = True
+            _LOGGER.exception(ex, 'Failed on sending a list of block payloads.')
         else:
             num_count += len(payload_block)
         finally:
-            # graceful exit
-            await device_client.shutdown()
+            # prevent to not execute when exception before client connection
+            if not exception_raised:
```

Comment:
What if exception raised after connection is established?

Reply:
exception raised only before connection made. And `.connect()` never raised any of exception. 
And this code is only to prevent the finally block execution.

Reply:
Well, My concern is

When `await device_client.connect()` is good but `await device_client.send_message(message)` got an exception, then?

Reply:
seems like we need nested try catch and disconnect the connection code repeat even on send_message exception.
But I haven't seen so far so let's tackle this separately in another JIRA or so.